packages/components/src/Card/Card.tsx

```diff
@@ -4,23 +4,55 @@ import { XOR } from "ts-xor";
 import styles from "./Card.css";
 import colors from "./colors.css";
 import { CardClickable } from "./CardClickable";
+import { Button, ButtonProps } from "../Button";
 import { Heading } from "../Heading";
 
-interface CardProps {
+interface CardCommonProps {
   /**
    * The `accent`, if provided, will effect the color accent at the top of
    * the card.
    */
   readonly accent?: keyof typeof colors;
   readonly children: ReactNode | ReactNode[];
+}
 
+interface HeaderActionProps {
   /**
    * The title of the card.
    */
   readonly title?: string;
+  /**
+   * The action props that renders into a button on the card header.
+   */
+  readonly action?: ButtonProps;
+}
+
+interface HeaderCustomProps {
+  /**
+   * The custom component to be rendered on the card header.
+   */
+  readonly customHeader?: React.ReactNode;
 }
 
-interface LinkCardProps extends CardProps {
+type HeaderProps = XOR<HeaderActionProps, HeaderCustomProps>;
+
+interface CardHeaderProps {
+  /**
+   * The header props of the card.
+   */
+  readonly header?: HeaderProps;
+}
+
+interface CardTitleProps {
+  /**
+   * @deprecated Use header instead
+   */
+  readonly title?: string;
+}
+
+type CardProps = CardCommonProps & XOR<CardHeaderProps, CardTitleProps>;
```

Comment:
Instead of doing all of this and XOR, have we thought about 

```ts
interface CardProps {
  ...
   /**
    * The title of the card.
    * 
    * @deprecated
    */
  readonly title?: string;

   /**
    * The header props of the card.
    */
  readonly header: string | HeaderProps | ReactNode
  /
}
```

that way it's always either one of those but not all of them. Don't need the XOR. On the code side, a classic `typeof` should do the trick.

Also simplifies the usage when you don't really need any of the advanced stuff and just wants a title.


```ts
<Card header="Bananas" />
<Card header="{ title: "Bananas", action: {...} }" />
<Card header="<MyBananaHeader />" />
```